crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs

```diff
@@ -0,0 +1,220 @@
+                // 1 - 2j
+                if let ExprKind::Constant {
+                    value: Constant::Int(..),
+                    ..
+                } = &left.node
+                {
+                    return checker.locator.slice(&Range::from_located(left)).len() <= 10;
+                } else if let ExprKind::UnaryOp {
+                    op: Unaryop::USub,
+                    operand,
+                } = &left.node
+                {
+                    // -1 + 2j
+                    // -1 - 2j
+                    if let ExprKind::Constant {
+                        value: Constant::Int(..),
+                        ..
+                    } = &operand.node
+                    {
+                        return checker.locator.slice(&Range::from_located(operand)).len() <= 10;
+                    }
+                }
+            }
+        }
+        // `sys.stdin`, etc.
+        ExprKind::Attribute { .. } => {
+            if checker
+                .resolve_call_path(default)
+                .map_or(false, |call_path| {
+                    ALLOWED_ATTRIBUTES_IN_DEFAULTS
+                        .iter()
+                        .any(|target| call_path.as_slice() == *target)
+                })
+            {
+                return true;
+            }
+        }
+        _ => {}
+    }
+    false
+}
+
+/// PYI011
+pub fn typed_argument_simple_defaults(checker: &mut Checker, args: &Arguments) {
+    if !args.defaults.is_empty() {
+        let defaults_start = args.posonlyargs.len() + args.args.len() - args.defaults.len();
+        for (i, arg) in args.posonlyargs.iter().chain(&args.args).enumerate() {
+            if let Some(default) = i
+                .checked_sub(defaults_start)
+                .and_then(|i| args.defaults.get(i))
+            {
+                if arg.node.annotation.is_some() {
+                    if !is_valid_default_value_with_annotation(default, checker) {
+                        checker.diagnostics.push(Diagnostic::new(
+                            TypedArgumentSimpleDefaults,
+                            Range::from_located(default),
+                        ));
+                    }
+                }
+            }
+        }
+    }
+
+    if !args.kw_defaults.is_empty() {
+        let defaults_start = args.kwonlyargs.len() - args.kw_defaults.len();
+        for (i, kwarg) in args.kwonlyargs.iter().enumerate() {
+            if let Some(default) = i
+                .checked_sub(defaults_start)
+                .and_then(|i| args.kw_defaults.get(i))
+            {
+                if kwarg.node.annotation.is_some() {
+                    if !is_valid_default_value_with_annotation(default, checker) {
+                        checker.diagnostics.push(Diagnostic::new(
+                            TypedArgumentSimpleDefaults,
+                            Range::from_located(default),
+                        ));
+                    }
+                }
+            }
+        }
+    }
```

Comment:
I changed the iteration logic here to mirror the way we match arguments to defaults in `generator.rs` -- namely, we need to take into account keywords, positional-only, etc.